### ISSUE TITLE
What to do for more than one html pages with common header, footer and Sidebar?

### ISSUE THREAD

**Issue by jitendravyas** [una/gulp-starter-env#10]

**Comment by una** [comment]
This is a starter without a templating environment because I just wanted to make a very basic prototyping environment (normally you'd use something like [handlebars](http://handlebarsjs.com/) or [lodash](https://lodash.com/docs#template) for templating with Javascript.

Here's an example of a gulp plugin that implements lodash: https://github.com/sindresorhus/gulp-template

I might make an alternative version of this to add templating. Do you think it should be included in the original? I was also going to turn it into a yeoman generator and that could potentially be one of the options. Decisions. decisions..

**Comment by jitendravyas** [comment]
I think yes having a template I can make HTML partials which help to keep the modular and only required file needs to be changed which is good for version control.

I like Mozilla's Nunjucks templating system.

**Comment by una** [comment]
I've never used Nunjucks but it looks interesting. I'm currently using [Dust](http://akdubya.github.io/dustjs/) on new projects.

If you want to implement Nunjucks and submit a PR i'd love to take a look (sounds like you're doing this anyway) -- this is something I'm a bit torn about adding to this starter environment

**Comment by una** [comment]
https://twitter.com/Una/status/641983072792125440 <-- linking a Twitter thread about this issue some people commented on

**Comment by onishiweb** [comment]
I think it depends a bit on what you want this to be a starter environment for. Having a templating environment adds an assumption that it will be used for a specific kind of site. Which is cool.

But Gulp is also really general and can be used on anything, i.e. if I were to use it on with WordPress, Laravel, or Rails for instance, I'd have no use for the templating and would either just have to remove it or leave it in as redundant. 

No harm in either, just depends on how general a "starter" env it's meant to be :)

**Comment by una** [comment]
I agree with you @onishiweb -- I feel like keeping i static is more welcoming to newcomers and also leaves it more open to be built upon. That being said, anything I generally make these days benefits from a templating engine. 

I'm sort of on the team for an option to include templating in a yeoman generator for this but leaving this repo one bare?
